src/vs/workbench/parts/files/common/explorerModel.ts

```diff
@@ -262,7 +265,8 @@ export class FileStat implements IFileStat {
 	}
 
 	private updateResource(recursive: boolean): void {
-		this.resource = URI.file(paths.join(this.parent.resource.fsPath, this.name));
+		this.resource = this.parent.resource.with({ path: paths.join(this.parent.resource.path, this.name) });
+		// this.resource = URI.file(paths.join(this.parent.resource.fsPath, this.name));
```

Comment:
I wonder how often the `URI.file`-thing occurs...

Reply:
Quick scan shows some ~50 uses (excluding tests). I think basically we have to check each and understand if the creator can work with any resource or really only files. E.g. if someone creates a URI only later to use some node FS APIs to work on the resource, that consumer is not ready for any other resource scheme.